apiml-security-common/src/main/java/com/ca/apiml/security/common/service/PassTicketService.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * This program and the accompanying materials are made available under the terms of the
+ * Eclipse Public License v2.0 which accompanies this distribution, and is available at
+ * https://www.eclipse.org/legal/epl-v20.html
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Copyright Contributors to the Zowe Project.
+ */
+package com.ca.apiml.security.common.service;
+
+import com.ca.mfaas.util.ClassOrDefaultProxyUtils;
+import lombok.AllArgsConstructor;
+import lombok.Value;
+import org.apache.commons.lang.StringUtils;
+import org.springframework.stereotype.Service;
+
+import javax.annotation.PostConstruct;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * This method allow to get a PassTicket from SAF.
+ */
+@Service
+public class PassTicketService {
+
+    private IRRPassTicket irrPassTicket;
+
+    @PostConstruct
+    public void init() {
+        this.irrPassTicket = ClassOrDefaultProxyUtils.createProxy(IRRPassTicket.class,
+                "com.ibm.eserver.zos.racf.IRRPassTicket", DefaultPassTicketImpl::new,
+                new ClassOrDefaultProxyUtils.ByMethodName<IRRPassTicketEvaluationException>(
+                        "com.ibm.eserver.zos.racf.IRRPassTicketEvaluationException",
+                        IRRPassTicketEvaluationException.class, "getSafRc", "getRacfRsn", "getRacfRc"),
+                new ClassOrDefaultProxyUtils.ByMethodName<IRRPassTicketGenerationException>(
+                        "com.ibm.eserver.zos.racf.IRRPassTicketGenerationException",
+                        IRRPassTicketGenerationException.class, "getSafRc", "getRacfRsn", "getRacfRc"));
+    }
+
+    public void evaluate(String userId, String applId, String passTicket) throws IRRPassTicketEvaluationException {
+        irrPassTicket.evaluate(userId, applId, passTicket);
+    }
+
+    public String generate(String userId, String applId) throws IRRPassTicketGenerationException {
+        return irrPassTicket.generate(userId, applId);
+    }
+
+    public boolean isUsingSafImplementation() {
```

Comment:
It looks like this method is not used anywhere

Reply:
Hi Andreas, thanks for a thorough review. You are right that this method is not used. We think that it is useful to have it here since it provides a way how to check if a real implementation is used.

Reply:
Hi Petr! That's fine, I was not sure if it was intentional. Thanks for the clarification!